alibi_detect/saving/tests/test_saving.py

```diff
@@ -406,12 +196,15 @@ def test_save_cvmdrift(data, preprocess_custom, tmp_path):
     ], indirect=True
 )
 @parametrize_with_cases("data", cases=ContinuousData, prefix='data_')
-def test_save_mmddrift(data, kernel, preprocess_custom, backend, tmp_path, seed):
+def test_save_mmddrift(data, kernel, preprocess_custom, backend, tmp_path, seed):  # noqa: F811
     """
     Test MMDDrift on continuous datasets, with UAE as preprocess_fn.
 
     Detector is saved and then loaded, with assertions checking that the reinstantiated detector is equivalent.
     """
+    if backend not in ('tensorflow', 'pytorch', 'keops'):
+        pytest.skip("Detector doesn't have this backend")
```

Comment:
Slightly confused about these statements, is this future-proofing if we had more backends? Also looking at the `backend` fixture it's only parametrized with `tensorflow` and `sklearn`, so how does it work for `keops`, `pytorch`?

Reply:
Yes future proofing, but more in the sense of future-proofing in anticipation of serialization support being extended. 

This file (`test_saving.py`) is currently only parametrized with `backend = param_fixture("backend", ['tensorflow', 'sklearn'])` because we currently only support saving with `tensorflow` and `sklearn`. As we add support for more backends, we should add them to this parametrization. 

The `if backend not in ('tensorflow', 'pytorch', 'keops'):` is intended to skip the test if any other `backend` is passed to it (e.g. atm `backend=='sklearn'` is passed), because `MMDDrift` only has `'tensorflow'`, `'pytorch'` and `'keops'` backends. There are various ways we could do this but I thought the above logic was the most maintainable because there is a one-to-one mapping between the contents of the tuple and the actual backends supported by the detector.